WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,20 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event)
+    event.id.in?(self.events.map(&:id))
+  end
+
+  def belongs_to_region?(region_or_country)
+    (self.countryId == region_or_country) || (self.country.continentId == region_or_country)
+  end
+
+  def search(search_param)
+    if search_param != nil
```

Comment:
Now I understood.
We will never call search (or contains) if the params are null, so no need to check it inside here.